server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java

```diff
@@ -557,6 +556,8 @@ public Username getRealName(String name) throws UsersRepositoryException {
 
     @Override
     public User getUserByName(Username name) throws UsersRepositoryException {
+        assertValid(name);
```

Comment:
I thought the goal here would be to check that in a virtual hosting case, we would check that the domain of the user is part of the domain list.... (more specifically in the ldap case). `assertValid` doesn't do that at all. 

Right check missing???

Reply:
If you look at the `AbstractUsersRepository.assertValid()` it does some proper checks depending on virtualHosting, including domainList check.

As know it inherit AbstractUsersRepository

Reply:
My bad I looked at `UsersRepository`